controllers/mirrorpeer_controller.go

```diff
@@ -339,7 +339,8 @@ func (r *MirrorPeerReconciler) checkForDestinationSecret(ctx context.Context, pe
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *MirrorPeerReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	mpPredicate := utils.ComposePredicates(predicate.GenerationChangedPredicate{})
+	// Reconcile will trigger if the generation changes or the annotations change
+	mpPredicate := predicate.Or(predicate.GenerationChangedPredicate{}, predicate.AnnotationChangedPredicate{})
```

Comment:
why do we need annotation change predicate to label resources?

Reply:
legacy code. removed